swap_server_client.go

```diff
@@ -58,10 +58,17 @@ func newSwapServerClient(address, proxyAddress string, insecure bool,
 	tlsPath string, lsatStore lsat.Store, lnd *lndclient.LndServices,
 	maxLSATCost, maxLSATFee btcutil.Amount) (*grpcSwapServerClient, error) {
 
+	// If a proxy address is set, we know tor is being used to secure the
+	// transport of our credentials (no other proxies are allowed
+	// currently). We need to set the flag accordingly, otherwise gRPC won't
+	// add the LSAT macaroon to the request.
+	allowInsecure := proxyAddress != ""
```

Comment:
This change isn't safe yet until we include the swap server onion addresses. When proxying over Tor to a clearnet HTTP server, the transport is not encrypted.

Reply:
Yes, you're right, forgot about that.